src/init.cc

```diff
@@ -823,6 +823,17 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, ncclUniqueId* comm
   NCCLCHECKGOTO(ncclTransportP2pSetup(comm, &treeGraph), ret, affinity_restore);
   INFO(NCCL_INIT, "Connected all trees");
 
+  NCCLCHECK(scklGetTopoFromXMLAndSetChannels(comm));
```

Comment:
Does everything work if you don't pass a sckl xml file?

Reply:
great question! Yes, it is guarded properly and in case it is given a bad path it throws a proper error.